### PR TITLE
Floor and Ceil Op

### PR DESCRIPTION
**Pull request by zem7**

### CODE REVIEW

File: caffe2/operators/ceil_op.cc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace caffe2 {
+namespace {
+REGISTER_CPU_OPERATOR(Ceil, CeilOp<float, CPUContext>);
+
+OPERATOR_SCHEMA(Ceil)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .AllowInplace({{0, 0}})
+    .SetDoc(R"DOC(
+Ceil takes one input data (Tensor<T>) and produces one output data
```
Comment (houseroad) [inline caffe2/operators/ceil_op.cc:30]: We probably should say that, this op only supports float32 for now.

File: caffe2/operators/ceil_op.cc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+(Tensor<T>) where the ceil function, y = ceil(x), is applied to
+the tensor elementwise.
+)DOC")
+    .Input(0, "X", "1D input tensor")
```
Comment (houseroad) [inline caffe2/operators/ceil_op.cc]: We should be more flexible. In ONNX, it supports higher dimension, not only 1D input tensor.

File: caffe2/operators/floor_op.cc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace caffe2 {
+namespace {
+REGISTER_CPU_OPERATOR(Floor, FloorOp<float, CPUContext>);
+
+OPERATOR_SCHEMA(Floor)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .AllowInplace({{0, 0}})
+    .SetDoc(R"DOC(
+Floor takes one input data (Tensor<T>) and produces one output data
```
Comment (houseroad) [inline caffe2/operators/floor_op.cc:30]: Same here, only float type is supported.

File: caffe2/operators/floor_op.cc [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+(Tensor<T>) where the floor function, y = floor(x), is applied to
+the tensor elementwise.
+)DOC")
+    .Input(0, "X", "1D input tensor")
```
Comment (houseroad) [inline caffe2/operators/floor_op.cc]: support higher dimension tensor?

**Review by houseroad — COMMENTED** [review]
Thanks! Please address the comments.

File: caffe2/operators/ceil_op.cc [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+OPERATOR_SCHEMA(Ceil)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .AllowInplace({{0, 0}})
+    .SetDoc(R"DOC(
+Ceil takes one input data (Tensor<T>) and produces one output data
+(Tensor<T>) where the ceil function, y = ceil(x), is applied to
+the tensor elementwise. Currently supports only float32.
+)DOC")
+    .Input(0, "X", "1D input tensor")
```
Comment (houseroad) [inline caffe2/operators/ceil_op.cc]: We support any dimension tensor, don't we?

File: caffe2/operators/floor_op.cc [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+OPERATOR_SCHEMA(Floor)
+    .NumInputs(1)
+    .NumOutputs(1)
+    .AllowInplace({{0, 0}})
+    .SetDoc(R"DOC(
+Floor takes one input data (Tensor<T>) and produces one output data
+(Tensor<T>) where the floor function, y = floor(x), is applied to
+the tensor elementwise. Currently supports only float32.
+)DOC")
+    .Input(0, "X", "1D input tensor")
```
Comment (houseroad) [inline caffe2/operators/floor_op.cc]: Also here, not only 1D tensor

**Comment by houseroad** [comment]
Also, please rebase your changes on the newest commit to let the CI pass.

**Comment by houseroad** [comment]
@caffe2bot retest this please

**Review by houseroad — APPROVED** [review]
First two ops are done. :-) Great job.
LGTM, thanks!

**Comment by houseroad** [comment]
Since we already landed our diff internally, let's close this PR and wait @orionr to manually sync the changes back to github. :-)

**Comment by orionr** [comment]
This is a good test! Thanks @houseroad and @zem7 - please verify your changes look good in https://github.com/caffe2/caffe2/pull/1996

**Comment by orionr** [comment]
@caffe2bot test this please

**Comment by orionr** [comment]
Landed with the sync from fbcode. Thanks all!

**Pull request closed without merging**